patches/server/0965-Make-banners-modifiable-with-both-BannerMeta-and-Blo.patch

```diff
@@ -0,0 +1,270 @@
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBanner.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBanner.java
+@@ -18,7 +18,7 @@ import org.bukkit.craftbukkit.util.CraftMagicNumbers;
+ import org.bukkit.inventory.meta.BannerMeta;
+ 
+ @DelegateDeserialization(CraftMetaItem.SerializableMeta.class)
+-public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
++public class CraftMetaBanner extends CraftMetaBlockState implements BannerMeta { // Paper - Make banners also modifiable with BlockStateMeta
+ 
+     private static final Set<Material> BANNER_MATERIALS = Sets.newHashSet(
+             Material.BLACK_BANNER,
+@@ -59,14 +59,29 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+     static final ItemMetaKey PATTERNS = new ItemMetaKey("Patterns", "patterns");
+     static final ItemMetaKey COLOR = new ItemMetaKey("Color", "color");
+     static final ItemMetaKey PATTERN = new ItemMetaKey("Pattern", "pattern");
++    // Paper start - Add banner materials to CraftMetaBlockState's material list
++    static {
++        CraftMetaBlockState.BLOCK_STATE_MATERIALS.addAll(CraftMetaBanner.BANNER_MATERIALS);
++    }
++    // Paper end - Add banner materials to CraftMetaBlockState's material list
+ 
+     private DyeColor base;
+     private List<Pattern> patterns = new ArrayList<Pattern>();
+-
++    // Paper start - Create new constructor to properly handle different banners
++    private final org.bukkit.block.Banner bannerBlock;
++    @Deprecated
+     CraftMetaBanner(CraftMetaItem meta) {
+-        super(meta);
++        this(meta, Material.WHITE_BANNER);
++
++        new org.bukkit.plugin.AuthorNagException("Usage of internal deprecated constructor in paper: CraftMetaBanner(CraftMetaItem), this will incorrectly assume the item is a white banner").printStackTrace();
++    }
++
++    CraftMetaBanner(CraftMetaItem meta, Material material) {
++        super(meta, material);
+ 
+-        if (!(meta instanceof CraftMetaBanner)) {
++        if (true || !(meta instanceof CraftMetaBanner)) {
++            this.bannerBlock = (org.bukkit.block.Banner) this.getBlockState();
++            // Paper end - Create new constructor to properly handle different banners
+             return;
+         }
+ 
+@@ -74,11 +89,20 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+         this.base = banner.base;
+         this.patterns = new ArrayList<Pattern>(banner.patterns);
+     }
+-
++    // Paper start - Create new constructor to properly handle different banners
++    @Deprecated
+     CraftMetaBanner(CompoundTag tag) {
+-        super(tag);
++        this(tag, Material.WHITE_BANNER);
++
++        new org.bukkit.plugin.AuthorNagException("Usage of internal deprecated constructor in paper: CraftMetaBanner(CompoundTag), this will incorrectly assume the item is a white banner").printStackTrace();
++    }
```

Comment:
Same here, just remove the constructor (or just add the Material param, same thing).